### PR TITLE
meson: add experimental bpf-gcc compiler support

### PR DESCRIPTION
**Pull request by jameshilliard**
Not fully working but should make it easier to clean up remaining issues.

### CODE REVIEW

File: src/core/bpf/meson.build [hunk 1]
```diff
@@ -13,6 +13,14 @@ clang_flags = [
         '-c',
```
Comment (yuwata) [inline src/core/bpf/meson.build:13]: Maybe, better to rename `clang_flags` -> `bpf_clang_flags` for consistency with `bpf_gcc_flags`?

File: src/core/bpf/meson.build [hunk 2]
```diff
@@ -30,19 +38,32 @@ cpu_arch_defines = {
         'arm'     : ['-D__arm__', '-D__ARM_PCS_VFP'],
 }
 
-clang_arch_flags = cpu_arch_defines.get(host_machine.cpu_family(),
+bpf_arch_flags = cpu_arch_defines.get(host_machine.cpu_family(),
                                         ['-D__@0@__'.format(host_machine.cpu_family())])
```
Comment (yuwata) [inline src/core/bpf/meson.build]: Indentation.

File: meson.build [hunk 3]
```diff
@@ -1013,58 +1013,72 @@ conf.set10('HAVE_LIBBPF', libbpf.found())
...
+                # Debian installs this in /usr/sbin/ which is not in $PATH.
+                # We check for 'bpftool' first, honouring $PATH, and in /usr/sbin/ for Debian.
+                # We use 'bpftool gen' subcommand, it was added by 985ead416df39d6fe8e89580cc1db6aa273e0175 (v5.6).
                 bpftool = find_program('bpftool',
                                        '/usr/sbin/bpftool',
-                                       required : bpf_framework_required,
-                                       version : '>= 5.6.0')
-        endif
+                                       required : false,
+                                       version : '>= 5.13.0')
```
Comment (yuwata) [inline meson.build:1065]: Could you update the comment why we need 5.13.0 here?

File: src/core/bpf/meson.build (hunk 1 above)
Comment (jameshilliard) [reply src/core/bpf/meson.build:13]: Changed as requested.

File: src/core/bpf/meson.build (hunk 2 above)
Comment (jameshilliard) [reply src/core/bpf/meson.build]: Fixed.

File: meson.build (hunk 3 above)
Comment (jameshilliard) [reply meson.build:1065]: Comments updated/added.

**Comment by yuwata** [comment]
> Not fully working but should make it easier to clean up remaining issues.

What feature not working now? Could you add a comment about the current limitation?

**Comment by jameshilliard** [comment]
> What feature not working now?

Seems to be mostly [libbpf issues](https://lore.kernel.org/bpf/20220609062412.3950380-1-james.hilliard1@gmail.com/) at the moment. I put a warning in that using this option is experimental and not recommended yet, it's mostly helpful to have build system support at the moment for reproducing/cleaning up the remaining issues in dependencies/tooling.

**Comment by yuwata** [comment]
Thanks. Then, LGTM.

**Review by yuwata — APPROVED** [review]

**Comment by keszybz** [comment]
focal-amd64 timed out in boot-and-services. This PR shouldn't have any effects after compilation, since the new option isn't being used, so this is unrelated.

**Pull request merged**